op-program/host/kvstore/disk.go

```diff
@@ -0,0 +1,64 @@
+package kvstore
+
+import (
+	"encoding/hex"
+	"fmt"
+	"io"
+	"os"
+	"path"
+
+	"github.com/ethereum/go-ethereum/common"
+)
+
+// read/write mode for user/group/other, not executable.
+const diskPermission = 0666
+
+// DiskKV is a disk-backed key-value store, every key-value pair is a hex-encoded .txt file, with the value as content.
+// DiskKV is safe for concurrent use; Puts may conflict, but write the exact same data anyway.
+type DiskKV struct {
+	path string
+}
+
+// NewDiskKV creates a DiskKV that puts/gets pre-images as files in the given directory path.
+// The path must exist, or subsequent Put/Get calls will error when it does not.
+func NewDiskKV(path string) *DiskKV {
+	return &DiskKV{path: path}
+}
+
+func (d *DiskKV) pathKey(k common.Hash) string {
+	return path.Join(d.path, k.String()+".txt")
+}
+
+func (d *DiskKV) Put(k common.Hash, v []byte) error {
+	// no O_EXCL, the pre-image may already exist. It's fine to overwrite it.
```

Comment:
```suggestion
	// no O_EXCL, the key file may already exist. It's fine to overwrite it.
```

since in this case it's the hash that already exists and we're overwriting its pre-image with a different value.

Reply:
see other comment, changed the behavior here, but thanks for the change suggestion